src/binary-writer.cc

```diff
@@ -1214,6 +1214,16 @@ Result BinaryWriter::WriteModule() {
       stream_->MoveData(data_count_start_, data_count_end_, size);
     }
     stream_->Truncate(data_count_start_ + size);
+
+    --section_count_;
+
+    // We just effectively decremented the code section's index; adjust anything
+    // that might have captured it.
+    for (RelocSection& section : reloc_sections_) {
+      if (section.section_index == section_count_) {
```

Comment:
I think this only works because the data count section comes immediately before the code section -- maybe we want to make that more explicit here? Or keep track of the data count section's index above and decrement any section_index that is after it.

Reply:
You are correct regarding the limitation, but it is also as specified: see https://github.com/WebAssembly/bulk-memory-operations/blob/master/proposals/bulk-memory-operations/Overview.md#datacount-section.  The data count section can only appear between the element and code sections.

Reply:
Good point, I guess I was concerned that this might break in a non-obvious way if we add a new section between the two. Maybe there's an assertion that could added?

Reply:
Sure, PTAL.  Thanks :)

Reply:
Great, thanks!